src/rest-server/src/models/v2/task.js

```diff
@@ -0,0 +1,127 @@
+//
+// MIT License
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+// documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+// to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+// The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+// BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+// module dependencies
+const logger = require('@pai/config/logger');
+const databaseModel = require('@pai/utils/dbUtils');
+const { encodeName } = require('@pai/models/v2/utils/name');
+const createError = require('@pai/utils/error');
+const {
+  convertToTaskDetail,
+} = require('@pai/models/v2/utils/frameworkConverter');
+
+const get = async (frameworkName, jobAttemptIndex, taskRoleName, taskIndex) => {
+  // get last task attempt by querying Framework / FrameworkHistory table
+  let attemptFramework;
+  const encodedFrameworkName = encodeName(frameworkName);
+
+  const framework = await databaseModel.Framework.findOne({
+    attributes: ['snapshot'],
+    where: { name: encodedFrameworkName },
+  });
+
+  if (framework) {
+    attemptFramework = JSON.parse(framework.snapshot);
+  } else {
+    throw createError(
+      'Not Found',
+      'NoJobError',
+      `Job ${frameworkName} is not found.`,
+    );
+  }
+
+  // when attemptIndex is not the last attempt, get the framework attempt from frameworkHistory table
+  if (jobAttemptIndex < attemptFramework.spec.retryPolicy.maxRetryCount) {
+    const historyFramework = await databaseModel.FrameworkHistory.findOne({
+      attributes: ['snapshot'],
+      where: {
+        frameworkName: encodedFrameworkName,
+        attemptIndex: jobAttemptIndex,
+      },
+    });
+
+    if (historyFramework) {
+      attemptFramework = JSON.parse(historyFramework.snapshot);
+    } else {
+      throw createError(
+        'Not Found',
+        'NoJobError',
+        `Job attempt not found in job ${frameworkName} with jobAttemptIndex ${jobAttemptIndex}.`,
+      );
+    }
+  } else if (
+    jobAttemptIndex > attemptFramework.spec.retryPolicy.maxRetryCount
+  ) {
+    throw createError(
+      'Not Found',
+      'NoJobError',
+      `Job attempt not found in job ${frameworkName} with jobAttemptIndex ${jobAttemptIndex}`,
+    );
+  }
```

Comment:
jobAttemptIndex is not comparable with maxRetryCount, as maxRetryCount only limits user failure retry count.
So change to:
If (jobAttemptIndex < attemptFramework.status.attemptStatus.id) {}
else if (jobAttemptIndex == attemptFramework.status.attemptStatus.id) {}
else {}

Reply:
> jobAttemptIndex is not comparable with maxRetryCount, as maxRetryCount only limits user failure retry count.
> So change to:
> If (jobAttemptIndex < attemptFramework.status.attemptStatus.id) {}
> else if (jobAttemptIndex === attemptFramework.status.attemptStatus.id) {}
> else {}

Fixed, thanks